### PR TITLE
Bio-Formats decoupling

### PR DESCRIPTION
**Pull request by sbesson**
This is the preliminary work allowing us to decouple Bio-Formats from OMERO as part of the build system. Instead of being included as a submodule and built every time, Bio-Formats is now managed as another dependency using Ivy. The Bio-Formats submodule is completely removed and the various targets are modified to allow all the OMERO components to build.

/cc @bramalingam @melissalinkert

### CODE REVIEW

**Comment by sbesson** [comment]
Given the status of https://trello.com/c/hQ7WeJ8w/29-testing and the https://ci.openmicroscopy.org/view/DEV queue:
- the full server/clients are now built without the Bio-Formats submodules
- the OMERO.server is successfully deployed and running
- integration tests are running: [Python](https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-integration-python/21/) is green and [Java](https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-integration-java/20/) still has failures/skips to resolve
- multiple file formats imports with no apparent regressions from OMERO 5.1.2

My proposal is to merge this epic PR and move forward with smaller topical PRs for:
- a full review of the dependencies state after the cleanup done by @bramalingam
- fixing the Java integration tests: ExporterTest need https://github.com/openmicroscopy/bioformats/pull/1836, need to investigate skipped tests

Any veto @joshmoore @melissalinkert @bramalingam @pwalczysko @jburel ?

**Comment by joshmoore** [comment]
None from me.

**Comment by jburel** [comment]
makes sense.no veto
wrt exporter. I will check if it is not quicker to move them to bf for now

File: etc/omero.properties [hunk 1]
```diff
@@ -911,7 +912,7 @@ versions.log4j=1.2.17
 versions.logback=1.1.1
 ## Change url in antlib/resources/release.xml when changing lucene version
 versions.lucene=2.4.1
-versions.mail=1.4.1
+versions.mail=1.4.4
```
Comment (sbesson) [inline etc/omero.properties]: This bump may have to be double-checked by the mail crew /cc @aleksandra-tarkowska @joshmoore

**Pull request merged**